protocol/adapters/datasync_client.go

```diff
@@ -0,0 +1,223 @@
+package adapters
```

Comment:
nitpick, but I would prefer the file to be named `client_datasync.go` if we go with this file structure